packages/loopring_v3/contracts/impl/Exchange.sol

```diff
@@ -427,7 +427,41 @@ contract Exchange is IExchange, Claimable, ReentrancyGuard
               revert(0, 0)
           }
         }
-        state.commitBlock(blockType, blockSize, blockVersion, decompressed, offchainData);
+        bytes32 publicDataHash = state.preCommit(blockType, decompressed, offchainData);
+
+        commitBlock(
+            blockType,
+            blockSize,
+            blockVersion,
+            publicDataHash,
+            decompressed
+        );
+
+        emit BlockCommitted(state.blocks.length + 1, publicDataHash);
+    }
+
+    function commitBlock(
+        uint8 blockType,
+        uint16  /* blockSize */,
+        uint8   /* blockVersion */,
+        bytes32 /* publicDataHash */,
+        bytes memory  /* data */
+        )
+        private
+    {
+        address _addr = state.loopring.getBlockProcessor(blockType);
+
+        assembly {
+            let ptr := mload(0x40)
+            calldatacopy(ptr, 0, calldatasize)
+            let result := delegatecall(gas, _addr, ptr, calldatasize, 0, 0)
+            let size := returndatasize
+            returndatacopy(ptr, 0, size)
+
+            switch result
+            case 0 { revert(ptr, size) }
+            default { return(ptr, size) }
+        }
```

Comment:
Delegatecall can also be done in [solidity](https://solidity.readthedocs.io/en/v0.5.10/units-and-global-variables.html?highlight=delegatecall#members-of-address-types) like this:
`<address>.delegatecall(bytes memory) returns (bool, bytes memory)`

`encodeWithSelector` can be used to encode the parameters in bytes

Reply:
This is not important, you can optimize it later :)